pkg/steps/project_image.go

```diff
@@ -64,6 +72,42 @@ func (s *projectDirectoryImageBuildStep) run(ctx context.Context) error {
 	return handleBuild(ctx, s.client, build)
 }
 
+func (s *projectDirectoryImageBuildStep) createSecrets() error {
+	logrus.Debugf("Creating test secrets used in build args for %q", s.Name())
+	if s.secretClient == nil {
+		return fmt.Errorf("cannot create secrets with a nil secret client")
+	}
+	toCreate := map[string]*coreapi.Secret{}
+	for i, args := range s.config.BuildArgs {
+		if args.ValueFrom != nil {
+			name, err := args.ValueFrom.SourceName()
+			if err != nil {
+				return fmt.Errorf("build_args[%d] failed to determine source name: %w", i, err)
+			}
+			raw := &coreapi.Secret{}
+			if err := s.secretClient.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: args.ValueFrom.Namespace, Name: args.ValueFrom.SecretKeyRef.Name}, raw); err != nil {
```

Comment:
Never use `context.TODO()`. All of our steps have `func (s *step) Run(ctx context.Context)`

Reply:
Done.